examples/preprocessing/interpolate_missing_channels.py

```diff
@@ -0,0 +1,93 @@
+"""
+==================================================================================
+Problem statement:
+sometimes a few MEG channels can be completely turned off during an experimental
+recording on purpose (i.e., bad sensors contain high noise).
+This is done inorder to stop noise spreading towards the other sensors.
+Therefore, physical location of the sensor is completely missing from
+the recording data. Therefore, the sensor can't be interpolated anymore.
+==================================================================================
+
+@author: diptyajit das <bmedasdiptyajit@gmail.com>
+Created on Jun 16, 2023
+
+"""
+
+# import some packages
+import mne
+from mne.datasets import sample
+from mne import read_evokeds
+import matplotlib.pyplot as plt
+from matplotlib import gridspec as grd
+
+print(__doc__)
+
+# define meg data path
+data_path = sample.data_path()
+evk_fname = data_path / "MEG" / "sample" / "sample_audvis-ave.fif"
+
+# read sample evoked data
+evk = read_evokeds(evk_fname)[0]
+
+# pick only grad channels and crop the data
+evk_grad = evk.copy().pick_types(meg='grad').crop(-0.05, 0.4)
+
+# correct baseline
+evk_grad.apply_baseline(baseline=(None, 0))
+
+# get the original channel names
+grad_ch_names = evk_grad.info['ch_names']
+
+# we specify one sensor (left auditory),
+# later we want to drop this sensor to create a missing sensor data file
+mark_channel = 'MEG 0212'
+pos = grad_ch_names.index(mark_channel)  # get sensor position
+
+# print the location // 'MEG 0212'
+print(evk_grad.info['chs'][pos]['loc'])
+
+# now lets drop the mark channel (in this case we choose a grad channel:'MEG 0212')
+# to create a missing channel scenario. this can be any bad sensor in practice
+mis_channel = 'MEG 0212'
+mis_channel_evk = evk_grad.copy().drop_channels(grad_ch_names[pos])
+
+# using MNE function to add an extra reference channel
+# it's a dummy reference channel ('flat' channel) inorder to retrieve our missing channel
+evk_recon = mne.add_reference_channels(mis_channel_evk, ref_channels=[mis_channel])
+
+# now change the channel type to grad
+evk_recon.set_channel_types({mis_channel: 'grad'})
+
+# check the info file that contains channel order
+# newly added channel name can be seen at the end of the channel order
+print('check the order of the channel names: ', evk_recon.info['ch_names'])
+
+# let's check our channel description that has the location of the sensor
+print('newly added channel location', evk_recon.info['chs'][-1]['loc'])
+
+# we replace the channel description with the original one
+evk_recon.info['chs'][-1] = evk_grad.info['chs'][pos]  # original the grad channel description
```

Comment:
We still need the original channel location from somewhere. `add_channels` only appends the channel. It doesn't add original channel location. So, at the moment I don't have any alternative solution to this.

Reply:
Is the use case actually for MEG data? If it's for EEG then we can use the `RawArray` approach plus `set_montage` on the one-channel array to set the position, which seems cleaner. I would adapt the example to do it for EEG, which seems like a *much* more prevalent case than MEG channels missing. (I've only seen MEG channels missing if someone setup the acquisition incorrectly at least.)

If it *must* be an MEG data example I would `.copy().pick(one_ch_name)` and use that as what gets `.add_channels`'ed. Because...

>  add_channels only appends the channel. It doesn't add original channel location.

... I don't think this part is correct. It will contain whatever channel information was present in the instance:
```
>>> import mne
>>> raw = mne.io.read_raw_fif(mne.datasets.sample.data_path() / 'MEG/sample/sample_audvis_raw.fif')
>>> raw.pick('grad')
<Raw | sample_audvis_raw.fif, 204 x 166800 (277.7 s), ~3.1 MB, data not loaded>
>>> raw_other = raw.copy().pick('MEG 0242')
>>> raw.drop_channels('MEG 0242')
<Raw | sample_audvis_raw.fif, 203 x 166800 (277.7 s), ~3.1 MB, data not loaded>
>>> raw.load_data()
<Raw | sample_audvis_raw.fif, 203 x 166800 (277.7 s), ~261.4 MB, data loaded>
>>> raw_other.load_data()
<Raw | sample_audvis_raw.fif, 1 x 166800 (277.7 s), ~4.2 MB, data loaded>
>>> raw.add_channels([raw_other])
<Raw | sample_audvis_raw.fif, 204 x 166800 (277.7 s), ~262.7 MB, data loaded>
>>> raw.info['chs'][-1]['ch_name']
'MEG 0242'
>>> raw.info['chs'][-1]['loc']
array([-0.1083    , -0.0011    ,  0.0071    ,  0.084999  , -0.99638599,
        0.0047    ,  0.067199  ,  0.0104    ,  0.99768502, -0.99412799,
       -0.084486  ,  0.06784   ])
```

Reply:
Ahh then in that case I would use the `.pick(...)` from an instance that *has* the sensor of interest to get the right channel information (as more or less done in the example above, except it used the same instance)

Reply:
> missing MEG channels is actually the motivating use case for @dasdiptyajit --- IIRC from the forum, they sometimes completely turn off MEG sensors due to high noise levels that spread to other sensors (@dasdiptyajit please confirm?)

Yes, this is correct. Its mainly for MEG data. I was looking at some of the old data in our lab and apparently, time to time different sensors (1-3) were turned off during recordings. Those channels were mainly corrupted with high local/environmental noises. Now, during computing a `grand average` maps, a lot of sensors are dropped. So, my idea was to reconstruct those missing channels and interpolate to avoid channel drop issue.

Reply:
> Ahh then in that case I would use the `.pick(...)` from an instance that _has_ the sensor of interest to get the right channel information (as more or less done in the example above, except it used the same instance)

I am gonna test this one. Thanks!

Reply:
> Ahh then in that case I would use the `.pick(...)` from an instance that _has_ the sensor of interest to get the right channel information (as more or less done in the example above, except it used the same instance)

-  `.pick(...)` works! It copies the physical location of the missing sensor. It can be a solution considering the user has a second dataset where the sensor of interest is present.
- What if the user doesn't have a second dataset? Is it possible to extract sensor location from any `BUILT-IN LAYOUTS`? I had a quick look, it's possible to get the sensor `pos` from the standard layout but they are `unit-normalized positions of the channels in 2d (x, y, width, height)` and don't translate the original sensor location. I would prefer to create a flat channel using `mne.EvokedArray` and update the channel location via `.info` from some built-in MEG layout which is commonly available.    

```
# create a flat grad channel to retrieve (i.e., reconstruct) the dropped channel
sampling_freq = evoked_drop.info['sfreq']
ch_types = "grad"
info = mne.create_info([drop_ch], ch_types=ch_types, sfreq=sampling_freq)
times = evoked.times
flat_ch_data = np.arange(len(times)).reshape(1, len(times))
flat_ch_data = np.zeros_like(flat_ch_data)
flat_evoked = mne.EvokedArray(flat_ch_data, info, tmin=times[0], nave=flat_ch_data.shape[0], comment="flat channel")
```

Reply:
> What if the user doesn't have a second dataset? Is it possible to extract sensor location from any `BUILT-IN LAYOUTS`?

Check the *montages* not the *layouts*.  Layouts are idealized 2D representations used only for plotting.

Reply:
> > Is it possible to extract sensor location from any BUILT-IN LAYOUTS

> Check the montages not the layouts. Layouts are idealized 2D representations used only for plotting.

Agreed layouts won't work. And I don't think montages are usable for MEG because MEG needs pos+ori and montages are really just sensor locations. I think the correct solution here is to have a second dataset that has the position of interest. After all you *must* have one if you're using `grand_average`, right?

Reply:
> > What if the user doesn't have a second dataset? Is it possible to extract sensor location from any `BUILT-IN LAYOUTS`?
> 
> Check the _montages_ not the _layouts_. Layouts are idealized 2D representations used only for plotting.

But Montages are for the `EEG`. Do you think I should extend the example for `EEG` too? Here we can use a standard montage and can simply use `set_montage` to update the channel location.

Reply:
> > > Is it possible to extract sensor location from any BUILT-IN LAYOUTS
> 
> > Check the montages not the layouts. Layouts are idealized 2D representations used only for plotting.
> 
> Agreed layouts won't work. And I don't think montages are usable for MEG because MEG needs pos+ori and montages are really just sensor locations. I think the correct solution here is to have a second dataset that has the position of interest. After all you _must_ have one if you're using `grand_average`, right?

Okey. If the purpose is only to do  `grand_average` then you are right. Its likely that use will have a second dataset.